instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/item/ItemInstrumenter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.item;
+
+import static io.opentelemetry.javaagent.instrumentation.spring.batch.SpringBatchInstrumentationConfig.instrumentationName;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import org.springframework.batch.core.scope.context.ChunkContext;
+
+public class ItemInstrumenter {
+
+  static final String ITEM_OPERATION_READ = "ItemRead";
+  static final String ITEM_OPERATION_WRITE = "ItemWrite";
+  static final String ITEM_OPERATION_PROCESS = "ItemProcess";
+
+  private static final Instrumenter<String, Void> INSTRUMENTER =
+      Instrumenter.<String, Void>newBuilder(
```

Comment:
This is a bit strange. There's no obvious `REQUEST` type for the instrumenter. The span name consists of data from the `ChunkContext` and a `String itemOperationName`. I could make a value class that combines these, but I figured since there is no attribute extraction I might as well use a `String` as the `REQUEST` type, and use `Function.identity()` (i.e. `str -> str`) as the span name extractor.

Reply:
Yeah that's fine - if we ever need to extract some attributes (I doubt we'll add any to the item-level spans - they're only here for the trace tree structure) we can always change that.